ethereum/eip712/encoding.go

```diff
@@ -219,3 +221,67 @@ func decodeProtobufSignDoc(signDocBytes []byte) (apitypes.TypedData, error) {
 
 	return typedData, nil
 }
+
+// validatePayloadMessages ensures that the transaction messages can be represented in an EIP-712
+// encoding by checking that messages exist, are of the same type, and share a single signer.
+func validatePayloadMessages(msgs []sdk.Msg) error {
+	if len(msgs) == 0 {
+		return errors.New("unable to build EIP-712 payload: transaction does contain any messages")
+	}
+
+	var msgType string
+	var msgSigner sdk.AccAddress
+
+	for i, m := range msgs {
+		t, err := getMsgType(m)
```

Comment:
rename variable `t` for readibility

Reply:
Any suggestions? Used `t` because it's easy to distinguish from `msgType`